Kernel/scheduler.c

```diff
@@ -0,0 +1,175 @@
+/* Local Headers */
+#include "time.h"
+#include "scheduler.h"
+#include "graphics.h"
+#include "memoryManager.h"
+
+#define MAX_PROCESSES 16
+#define QUANTUM 4
+#define PROCESS_STACK_SIZE 4096  
+
+typedef enum status {READY = 0, BLOCKED, KILLED} TprocessStatus;
+
+typedef struct processContext {
+    uint8_t priority;
+    TprocessStatus status;
+    uint64_t * topRSP;
+    uint64_t * baseRSP;
```

Comment:
Is this a pointer to end of the stack? In case of an affirmative answer, could you please tell me when are you planning to use or modify this pointer? As far as i can understand by reading the code, i think it can always be obtained with the top of the stack and the stack size already defined. 

And just to clarify, the top of the stack is necessary to store the allocated memory for the stack and then make a free once the process ends and the current rsp is going to be used when switching contexts. Is this correct?